exporter/fileexporter/file_exporter.go

```diff
@@ -72,10 +73,21 @@ func exportMessageAsLine(e *fileExporter, message proto.Message) error {
 func (e *fileExporter) Start(context.Context, component.Host) error {
 	var err error
 	e.file, err = os.OpenFile(e.path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
+	if err == nil {
+		e.isFileClosed = false
+	}
 	return err
 }
 
 // Shutdown stops the exporter and is invoked during shutdown.
 func (e *fileExporter) Shutdown(context.Context) error {
-	return e.file.Close()
+	if e.isFileClosed {
```

Comment:
It seems you are adding this to prevent calling `file.Close()` more than once, it _may_ be a good option to use the lock to do this. Not sure though, and I will leave that for the maintainers to decide ;)